cmd/register/main.go

```diff
@@ -132,20 +141,42 @@ func newCommand(fs vfs.FS, client register.Client, stateHandler register.StateHa
 						return fmt.Errorf("Installing local agent config")
 					}
 				} else {
+					log.Info("Retrieving Network configuration")
+					networkConfigData, err := client.GetNetworkConfig(cfg.Elemental.Registration, caCert, &registrationState)
```

Comment:
Don't we already have `networkConfigData` as part of `data` (and yaml unmarshalled `cfg`) above?
`data` should be the full MachineRegistration elemental config, including the networking part (if not there maybe we have to explicitly add on the operator side).
We may save extending the websocket protocol  here

Reply:
The `data` contains the `MachineRegistration` data, but this will not contain the real `IPAddresses`.

I want to avoid replacing this data (and wait for real IPs) at this stage, this is why I think it's better to have a more explicit request. 
Extending the protocol is not a problem I believe, but definitely we can improve in the future by not closing the websocket connection.